From c178b97f069e9c5f183039ba8892b68c35770691 Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Tue, 17 Sep 2013 17:46:18 +0000
Subject: [PATCH 0478/1076] massai_systools.lib V1.0.0.11

Mantis 0029028: mstarter: if process cannot be started, the pid 0 is used to find subprocesses
  ignore subprocesses if a processes cannot be started


git-svn-id: svn://localhost/SelfServiceCommon/trunk@589 90b65887-3827-0410-9a23-83215b262276
---
 SelfServiceCommon/Massai/cpp/Systools/src/local_def.cpp    | 7 ++++++-
 .../Massai/cpp/Systools/src/mSysProcessListW32.cpp         | 5 ++++-
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/SelfServiceCommon/Massai/cpp/Systools/src/local_def.cpp b/SelfServiceCommon/Massai/cpp/Systools/src/local_def.cpp
index 5d102ee0..1f836096 100644
--- a/SelfServiceCommon/Massai/cpp/Systools/src/local_def.cpp
+++ b/SelfServiceCommon/Massai/cpp/Systools/src/local_def.cpp
@@ -1,12 +1,17 @@
 #include "Tools/VERSIONLIB.h"
 
-VERSION_ID_DT(systools, 1, 0, 0, 10);
+VERSION_ID_DT(systools, 1, 0, 0, 11);
 
 /*!
 \page massai_systools.lib, massai_systools.lib - Release Notes
 
 massai_systools.lib
 
+\date 17.09.2013
+\li V1.0.0.11
+\li Mantis 0029028: mstarter: if process cannot be started, the pid 0 is used to find subprocesses
+  ignore subprocesses if a processes cannot be started
+
 \date 16.09.2013
 \li V1.0.0.10
 \li Mantis 0028903: MR: MCR Sankyo ICM 350
diff --git a/SelfServiceCommon/Massai/cpp/Systools/src/mSysProcessListW32.cpp b/SelfServiceCommon/Massai/cpp/Systools/src/mSysProcessListW32.cpp
index 8f054e26..9ff1e951 100644
--- a/SelfServiceCommon/Massai/cpp/Systools/src/mSysProcessListW32.cpp
+++ b/SelfServiceCommon/Massai/cpp/Systools/src/mSysProcessListW32.cpp
@@ -766,7 +766,10 @@ namespace
       ProcessHolderPtrList::const_iterator end = m_childs.end();
       for(; it != end; it++)
       {
-        check.push_back(it->get());
+        if((*it)->getProcess()->getStatus() != STATUS_FAILED)
+        {
+          check.push_back(it->get());
+        }
       }
     }
     {
-- 
2.41.0.windows.1

